gc.c

```diff
@@ -977,7 +1093,18 @@ static __inline__ tick_t
 tick(void)
 {
     unsigned long hi, lo;
-    __asm__ __volatile__ ("rdtsc" : "=a"(lo), "=d"(hi));
+    int lfence_before_rdtsc = how_serialize_rdtsc();
+    if (lfence_before_rdtsc) {
+        __asm__ __volatile__ (
+            "lfence\n\t"
+            "rdtsc\n\t"
+            : "=a"(lo), "=d"(hi));
+    } else {
+        __asm__ __volatile__ (
+            "mfence\n\t"
+            "rdtsc\n\t"
+            : "=a"(lo), "=d"(hi));
+    }
     return ((unsigned long long)lo)|( ((unsigned long long)hi)<<32);
```

Comment:
don't you try `rdtscp` ? ;)

Reply:
`rdtscp` returns a different value per cpu core.
see also http://proc-cpuinfo.fixstars.com/2014/11/rdtscp-html/
I think just rdtsc sounds reasonable as you wrote on [your article](https://qiita.com/kubo39/items/4319fa243fd18acc0981).

Reply:
I agree, " let's complicate it step-by-step".